### PR TITLE
[16.0][IMP] contract and contract_sale_generation: Add translations

### PR DESCRIPTION
**Pull request by aliciagaarzo**
The 'contract' and 'contract_sale_generation' modules have been modified so that the generation type selection values are translatable.

### CODE REVIEW

File: contract/models/abstract_contract.py [hunk 1]
```diff
@@ -51,15 +51,11 @@ class ContractAbstractContract(models.AbstractModel):
...
     generation_type = fields.Selection(
-        selection=lambda self: self._selection_generation_type(),
+        selection=[("invoice", "Invoice")],
         default=lambda self: self._default_generation_type(),
         help="Choose the document that will be automatically generated by cron.",
     )
 
-    @api.model
-    def _selection_generation_type(self):
-        return [("invoice", "Invoice")]
```
Comment (amarcosg) [inline contract/models/abstract_contract.py:61]: Deleting the function you may break other funcionality that depends on it. Dont delete de function, just add the _ in the option

```
    @api.model
    def _selection_generation_type(self):
        return [("invoice", _("Invoice"))]
```

File: contract_sale_generation/models/abstract_contract.py [hunk 2]
```diff
@@ -2,16 +2,14 @@
...
+from odoo import fields, models
 
 
 class ContractAbstractContract(models.AbstractModel):
     _inherit = "contract.abstract.contract"
 
     sale_autoconfirm = fields.Boolean()
 
-    @api.model
-    def _selection_generation_type(self):
```
Comment (amarcosg) [inline contract_sale_generation/models/abstract_contract.py:14]: Same dont delete the function just add the _ `_("Sale")`

**Review by amarcosg — CHANGES_REQUESTED** [review]
With the requested changes you may need to reexport the translations

File: contract_sale_generation/models/abstract_contract.py (hunk 2 above)
Comment (amarcosg) [reply contract_sale_generation/models/abstract_contract.py:14]: I tried to workaround it but the problem seems related to Selection translations in Abstract model, which is the case. https://github.com/odoo/odoo/issues/64065

Taking that in account for me the change is OK

**Review by amarcosg — APPROVED** [review]
LGTM

**Comment by flalexg** [comment]
Functional Review. LGTM.
image
image

**Review by flalexg — APPROVED** [review]

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge minor

**Pull request merged**